src/api/objects/ObjectAPI.js

```diff
@@ -230,11 +230,19 @@ ObjectAPI.prototype.get = function (identifier, abortSignal) {
  *          each resolving to domain objects matching provided search query and options.
  */
 ObjectAPI.prototype.search = function (query, abortSignal) {
+    return this.queryUsingProvider('searchForObjects', query, abortSignal);
```

Comment:
using the new `queryUsingProvider`, but leaving `search` for backwards compatibility

Reply:
This makes sense, I see the need for differentiating between different types of search. 

I wonder if it would make the interface simpler (and retain backwards compatibility) if we optionally passed the search type into the `search` function and retained that interface (and defaulted it to object search for backwards compatibility)? 

Also, using a "supportsSearch" type function to determine which providers can provide results might be a little more consistent with how the rest of our providers work in our API.

I'm also thinking we define the search function types in a const.

eg for all of the above.

``` JavaScript
this.SEARCH_TYPES = Object.freeze({
    OBJECT: 'OBJECT_SEARCH',
});

ObjectAPI.prototype.search = function (query, abortSignal, searchType=this.SEARCH_TYPES.OBJECT) {
    const providers = await this.getSearchProviders(searchType);
    const results = await Promise.all(providers.map(provider => provider.search(query, abortSignal, searchType)));
    
    return results.flat();
}
ObjectAPI.prototype.getSearchProviders(searchType) {
    return Object.values(this.providers).filter(provider => provider.supportsSearchType(searchType));
}
```